assets/src/hooks/useNearestIntersection.ts

```diff
@@ -1,13 +1,60 @@
-import { useEffect, useState } from "react"
+import { useEffect, useRef, useState } from "react"
 import { fetchNearestIntersection } from "../api"
+import {
+  LoadingResult,
+  isOk,
+  isLoadingState,
+  Err,
+  Ok,
+  Loading,
+} from "../helpers/loadingResult"
 
-export const useNearestIntersection = (
+export const useNearestIntersectionResult = (
   latitude: number,
   longitude: number
-): string | null => {
-  const [result, setResult] = useState<string | null>(null)
+): LoadingResult<string> => {
+  const latLong = { latitude, longitude }
+  const lastLatLong = useRef(latLong)
+  const [loadingState, setLoadingState] = useState<LoadingResult<string>>(
+    Loading()
+  )
+
   useEffect(() => {
-    fetchNearestIntersection(latitude, longitude).then(setResult)
+    setLoadingState(Loading())
+    ;(async () => {
+      const intersectionResult = await fetchNearestIntersection(
+        latitude,
+        longitude
+      )
+      if (intersectionResult !== null) {
+        setLoadingState(Ok(intersectionResult))
+      } else {
+        setLoadingState(
+          Err(new Error("`fetchNearestIntersection` returned `null`"))
+        )
+      }
+    })()
   }, [latitude, longitude])
-  return result
+
+  if (
+    lastLatLong.current.latitude !== latLong.latitude ||
+    lastLatLong.current.longitude !== latLong.longitude
+  ) {
+    lastLatLong.current = latLong
+    setLoadingState(Loading()) // redundant but needed to keep state synced?...
+    return Loading()
+  } else {
+    return loadingState
+  }
+}
+
+export const useNearestIntersection = (
+  latitude: number,
+  longitude: number
+): string | null => {
+  const statusResult = useNearestIntersectionResult(latitude, longitude)
+  if (!isLoadingState(statusResult) && isOk(statusResult)) {
+    return statusResult.ok
+  }
+  return null
```

Comment:
**question:** if I understand correctly, this means that once the latitude and longitude change but before the new result comes in the hook will return `null`, correct? What will this do to existing consumers (mainly the VPP)?

Reply:
oh! I meant to preserve old functionality, but I was going off only tests. I wasn't expecting to impact other areas because I don't have the scope to test/write tests for it.

I'm thinking of caching it and returning the old value till the next state, unless you think this is better and the usages should be updated instead.